### PR TITLE
fix(realtime): add onStart and onStop callback options for RealtimeLayer

### PR DESCRIPTION
**Pull request by danji90**
# How to
Add callbacks for tracking start and stop in RealtimeLayer

# Others

- [ ] It's not a hack or at least an unauthorized hack :).
- [ ] The images added are optimized.
- [ ] Everything in ticket description has been fixed.
- [ ] The author of the MR has made its own review before assigning the reviewer.
- [ ] The title means something for a human being and follows the [conventional commits](https://www.conventionalcommits.org/) specification.
- [ ] The title contains [WIP] if it's necessary.
- [ ] Labels applied. if it's a release? a hotfix?
- [ ] The new class' members & methods are well documented
- [ ] Tests added.

### CODE REVIEW

File: src/common/mixins/RealtimeLayerMixin.ts [hunk 1]
```diff
@@ -86,7 +89,7 @@ export type RealtimeLayerMixinOptions = OlLayerOptions & {
 /**
  * RealtimeLayerInterface.
  */
-export class RealtimeLayerInterface {
```
Comment (friedjoff) [inline src/common/mixins/RealtimeLayerMixin.ts:89]: The new name sounds a bit generic to me ;)

File: src/common/mixins/RealtimeLayerMixin.ts [hunk 2]
```diff
@@ -212,6 +215,10 @@ function RealtimeLayerMixin<T extends AnyLayerClass>(Base: T) {
       noInterpolate?: boolean,
     ) => void;
 
+    onStart?: (realtimeLayer: AnyLayer) => void;
```
Comment (friedjoff) [inline src/common/mixins/RealtimeLayerMixin.ts:218]: Can't we use `AnyRealtimeLayer` here?

**Review by friedjoff — CHANGES_REQUESTED** [review]
Seems like a good idea, but naming and typing looks a bit unclear to me.

File: src/common/mixins/RealtimeLayerMixin.ts (hunk 1 above)
Comment (danji90) [reply src/common/mixins/RealtimeLayerMixin.ts:89]: Opps, no idea how that happened

**Pull request merged**